website/docs/guides/data_source_filters.html.markdown

```diff
@@ -0,0 +1,286 @@
+you know that there is only one such entity**, and you want to access it without knowing the name.
+Like populated filters, when the search returns more than one item, the data source retrieval fails.
+
+### About metadata search
+
+Metadata can be searched even for those data sources that don't expose metadata in their interface. If the `filter`
+section lists `metadata` among the available criteria, you can search the metadata and get the results accordingly,
+although the vcd provider may not show the metadata for the found item.
+Note that the names of the metadata fields are case sensitive.
+
+## Example filter 1
+
+```hcl
+data "vcd_catalog_item" "unknown" {
+
+  org     = "datacloud"
+  catalog = "cat-datacloud"
+  
+  filter {
+    name_regex = "^p.*11$"
+  }
+}
+
+output "filtered_item" {
+    value = data.vcd_catalog_item.unknown
+}
+```
+
+Will find a catalog item with name starting with `p` and ending with `11`.
+It fails if there are several items named as requested, such as `photon-v11`, `platform911`, or `poorName211`
+Note that regular expressions are case sensitive: `photon-v11` and `Photon-v11` are two different entities.
+
+## Example filter 2
+
+```hcl
+data "vcd_catalog_item" "unknown" {
+
+  org     = "datacloud"
+  catalog = "cat-datacloud"
+  
+  filter {
+    name_regex = "^CentOS"
+    latest     = true
+  }
+}
+```
+
+Will find the most recent item where the name starts by `CentOS`.
+
+## Example filter 3
+
+```hcl
+data "vcd_catalog_item" "unknown" {
+
+  org     = "datacloud"
+  catalog = "cat-datacloud"
+  
+  filter {
+    date   = ">= 2020-03-20"
+    latest = true
+  }
+}
+# Alternative date conditions for the same value:
+# date = ">= March 23rd, 2020"
+# date = ">= 22-Mar-2020"
+# date = ">= 22-Mar-2020 00:00:00"
+# date = ">= 2020-03-23 00:00:00"
+```
+
+Will find the most recent item created on or after March 23rd, 2020
+
+## Example filter 4
+
+```hcl
+data "vcd_catalog_item" "unknown" {
+
+  org     = "datacloud"
+  catalog = "cat-datacloud"
+
+  filter {
+    metadata {
+     key   = "Code"
+     value = "BlackEagle"
+    }
+  }
+}
+```
+
+Finds an item where the metadata contains a key `Code` with value `BlackEagle`.
+Will fail if the criteria match more than one item.
+
+## Example filter 5
+
+```hcl
+data "vcd_catalog_item" "unknown" {
+
+  org     = "datacloud"
+  catalog = "cat-datacloud"
+
+  filter {
+    metadata {
+     key            = "ONE"
+     value          = "FirstValue"  # explicit value
+     use_api_search = "true"
+     type           = "STRING"
+    }
+    metadata {
+     key            = "TWO"
+     value          = "SecondValue" # explicit value
+     use_api_search = "true"
+     type           = "STRING"
+    }
+  }
+}
+```
+
+You can use several `metadata` blocks. This example finds an item where the metadata contains a key `ONE` with
+value `FirstValue`, and a key `TWO` with value `SecondValue`.
```

Comment:
Do I understand right that it's only metadata that supports using several conditions at once? If so, we need to emphasize that this is the logical AND operation. So, at the very least something like this or maybe you have some brighter ideas where to write it up:
```suggestion
value `FirstValue` AND a key `TWO` with value `SecondValue`.
```

Reply:
Yes. Correct. I enhanced the wording here and I added a section in the docs about multiple filter clauses